src/tsn_adapters/blocks/tn_access.py

```
@@ -599,37 +599,148 @@ def bulk_insert_tn_records(
+        for row in records.itertuples(index=False):
+            date_int = int(row.date)
+            if not check_unix_timestamp(date_int):
+                raise ValueError(
+                    f"date {row.date} for stream {row.stream_id} is not a unix timestamp"
+                )
+            flat_batches.append({
+                "stream_id": str(row.stream_id),
+                "inputs": [{"date": date_int, "value": float(row.value)}],
+            })
+
+        # Resume budget. 5 in-process attempts × 30s pause covers ~2.5 min of
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)